sdk/core/Azure.Core.Experimental/src/RequestOptions.cs

```diff
@@ -52,6 +97,20 @@ public class RequestOptions
         /// </summary>
         public HttpPipelinePolicy? PerCallPolicy { get; set; }
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="message"></param>
+        public void Apply(HttpMessage message)
+        {
+            if (PerCallPolicy != null)
+            {
+                message.SetProperty("RequestOptionsPerCallPolicyCallback", PerCallPolicy);
+            }
+
+            message.ResponseClassifier = new PerCallResponseClassifier(message.ResponseClassifier, this);
```

Comment:
Can we not allocate here? My prototype was no allocating if options were not consequential.

Reply:
Yep, we can.